Dev build for Cantera 3.2.0a3

speth opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Notable changes:
- Specify use of OpenBLAS instead of Intel MKL on Windows. Tests of these packages were failing with unexpected integration errors that I believe are related to incompatibilities between MKL and AMD processors on Windows that Intel can't be bothered to fix. For reference, reports of these problems on the Cantera Users' group [here](https://groups.google.com/g/cantera-users/c/9qZgoAuCQrg/m/AeH1TE1lAAAJ) and [here](https://groups.google.com/g/cantera-users/c/M_g4RINP_FY/m/LhAg5_8oDAAJ).
- Update pinned package versions. Once we merge the changes here into the `main` branch for the final Cantera 3.2.0 release, we can close #56, #58, #59, #61, and #62.

regro-cf-autotick-bot committed:
Rebuild for hdf5 1.14.4 (and keep 1.14.3)

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.44.9, and conda-forge-pinning 2024.12.06.19.20.35

bryanwweber committed:
Remove duplicate build section

bryanwweber committed:
Merge pull request #53 from regro-cf-autotick-bot/rebuild-hdf51144-0-2_h625c04

Rebuild for hdf5 1.14.4 (and keep 1.14.3)

speth committed:
Build Cantera 3.1.0

speth committed:
Remove obsolete patches

speth committed:
Pull build changes from dev branch

speth committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.45.0, and conda-forge-pinning 2024.12.16.17.14.48

speth committed:
Fix warning from conda-forge-tick

speth committed:
Relax build version restriction for sundials

bryanwweber committed:
Merge pull request #54 from speth/release-3.1.0

Release 3.1.0

speth committed:
Bump build number

bryanwweber committed:
Merge pull request #55 from speth/rebuild-3.1.0

Bump build number

regro-cf-autotick-bot committed:
Rebuild for fmt 11.2

regro-cf-autotick-bot committed:
Rebuild for libboost 1.88

regro-cf-autotick-bot committed:
Rebuild for python 3.14

speth committed:
Build Cantera 3.2.0a3

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

speth committed:
Merge branch 'main' into 'dev'

speth committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.08.08.18.38

Other tools:
- conda-build 24.9.0
- rattler-build 0.29.0
- rattler-build-conda-compat 1.4.6

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17803249965](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17803249965). Examine the logs at this URL for more detail.</sub>

speth committed:
Add build dependencies for sourcegen-generated CLib

speth committed:
Make numpy pin explicit

speth committed:
Delete maybe-obsolete Numpy 2 migration and rerender

speth committed:
Update for Cantera 3.2.0a3

Incorporate fix to allow use of Python 3.14

speth commented:
On all the Windows builders, the `openmp_ignition` examples is now failing as follows:
```
(%PREFIX%) %PREFIX%\share\cantera\samples\cxx\openmp_ignition>mkdir build 

(%PREFIX%) %PREFIX%\share\cantera\samples\cxx\openmp_ignition>cd build 

(%PREFIX%) %PREFIX%\share\cantera\samples\cxx\openmp_ignition\build>cmake .. 
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
-- Building for: Visual Studio 17 2022
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Selecting Windows SDK version 10.0.26100.0 to target Windows 10.0.20348.
-- The C compiler identification is MSVC 19.44.35215.0
-- The CXX compiler identification is MSVC 19.44.35215.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.44.35207/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.44.35207/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Found Threads: TRUE
-- Found OpenMP_C: -openmp (found version "2.0")
-- Found OpenMP_CXX: -openmp (found version "2.0")
-- Found OpenMP: TRUE (found version "2.0")
-- Configuring done (7.5s)
-- Generating done (0.0s)
-- Build files have been written to: D:/bld/cantera-recipe_1757437989973/_test_env/share/cantera/samples/cxx/openmp_ignition/build

(%PREFIX%) %PREFIX%\share\cantera\samples\cxx\openmp_ignition\build>cmake --build . --config Release 
MSBuild version 17.14.19+164abd434 for .NET Framework

  1>Checking Build System
  Building Custom Rule D:/bld/cantera-recipe_1757437989973/_test_env/share/cantera/samples/cxx/openmp_ignition/CMakeLists.txt
  Microsoft (R) C/C++ Optimizing Compiler Version 19.44.35215 for x64
  Copyright (C) Microsoft Corporation.  All rights reserved.
  cl /c /W3 /WX- /diagnostics:column /O2 /Ob2 /D _MBCS /D WIN32 /D _WINDOWS /D NDEBUG /D FMT_SHARED /D "CMAKE_INTDIR=\"Release\"" /EHsc /MD /GS /fp:precise /GR /openmp /std:c++17 /Fo"openmp_ignition.dir\Release\\" /Fd"openmp_ignition.dir\Release\vc143.pdb" /external:W0 /Gd /TP /errorReport:queue  /external:I "D:/bld/cantera-recipe_1757437989973/_test_env/Library/include" /utf-8 "%PREFIX%\share\cantera\samples\cxx\openmp_ignition\openmp_ignition.cpp"
  openmp_ignition.cpp
  openmp_ignition.vcxproj -> %PREFIX%\share\cantera\samples\cxx\openmp_ignition\build\Release\openmp_ignition.exe
  Building Custom Rule D:/bld/cantera-recipe_1757437989973/_test_env/share/cantera/samples/cxx/openmp_ignition/CMakeLists.txt

(%PREFIX%) %PREFIX%\share\cantera\samples\cxx\openmp_ignition\build>Release\openmp_ignition.exe

(%PREFIX%) %PREFIX%\share\cantera\samples\cxx\openmp_ignition\build>if -1073740791 NEQ 0 exit 1 
```

A [StackOverflow](https://stackoverflow.com/questions/50562192/process-finished-with-exit-code-1073740791-0xc0000409-pycharm-error) post suggests that the `-1073740791` is an exception triggered by some check indicated by the `/GS` compiler flag. However, this doesn't seem to provide any useful information about _where_ the exception is coming from. Hopefully I can reproduce this on a local Windows build.

speth commented:
I was able to build Cantera and this particular example locally, using the exact same version of the MS compiler (19.44.35215) and the example runs without error. I'm at a loss on how to debug this failure further.

speth committed:
Adjust sample program tests on Windows

speth committed:
Try pinning Windows to use OpenBLAS

speth committed:
Incorporate update to generated CMakeLists.txt

speth committed:
Incorporate update to clib sample install path

speth committed:
Incorporate fix for clib header installation

speth committed:
Avoid deprecated functions in extensible rate example

bryanwweber reviewed (approved):
Thanks for all this work Ray!
